HW06.Rmd

```diff
@@ -1,92 +1,136 @@
+#install.packages("XML", repos = "http://www.omegahat.net/R")
+library(XML)
+library(BiocManager)
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+
+GBM.transcripts <- full_list
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+GBM.genes.of.interest <- gather(GBM.genes.of.interest, key = "sample", value = "TPM", 2:46)%>%
+  separate(col = sample, into = c("Sample","Number"),sep = "\\.")
+for (i in genes_of_interest){
+  gene <- filter(GBM.genes.of.interest, gene_symbol == i)
+  print(i)
+  print(t.test(gene$TPM[gene$Sample == "H3K27M"], gene$TPM[gene$Sample == "WILDTYPE"]))
+}
+#print out the t-test results
+```
+
+
+
+Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
+
+```{r create a graphing function}
+#to work in the tidyverse, it will be easier to make tidy the dataframe first
+#create a graphing function
+#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
+
+for (i in genes_of_interest){
+  filtered <- filter(GBM.genes.of.interest, gene_symbol == i)
+  figure <- ggplot(filtered, aes(Sample, TPM))+
+    geom_boxplot(aes(fill = Sample))+
+    labs(title = paste(i, "Expression in GBM models by H3K27 Mutated or WT status"))
+  print(figure)
```

Comment:
Your code of creating boxplots is both concise and effective! In order to better simulate the example, you might want to center the title and remove legend from the graph.